File: 단계별로 풀어보기/반복문/Python/1110.py
# 1110
# 더하기 사이클
# 브론즈1

import sys

n = int(sys.stdin.readline())
li = []
tmp = 0
cnt = 0

if( n < 10 ):
    li.append(0)
    li.append(n)
else:
    li.append(n//10)
    li.append(n%10)
while(True):
    cnt += 1
    tmp = li[0] + li[1]
    tmp = tmp%10
    del li[0]
    li.append(tmp)
    if ( n == (li[0]*10 + li[1])):
        print(cnt)
        break
    else:
        continue

File: 단계별로 풀어보기/반복문/C++/1110.cpp
// 1110
// 더하기 사이클
// 브론즈1

#include <stdio.h>

int main(){
    int n = 0;
    int tmp = 0;
    int cnt = 0;
    int li[2];

    scanf("%d", &n);

    if (n<10){
        li[0] = 0;
        li[1] = n;
    }
    else{
        li[0] = n/10;
        li[1] = n%10;
    }
    while(1){
        cnt += 1;
        tmp = li[0] + li[1];
        tmp = tmp%10;
        li[0] = li[1];
        li[1] = tmp;
        if(n == (li[0]*10 + li[1])){
            printf("%d", cnt);
            break;
        }
        else{
            continue;
        }
    }
}